### PR TITLE
Improving the XML doc for RedirecUri (following-up customer feedback)

### PR DESCRIPTION
**Pull request by jmprieur**

### CODE REVIEW

**Review by bgavrilMS — APPROVED** [review]

File: src/client/Microsoft.Identity.Client/AppConfig/ApplicationOptions.cs [hunk 1]
```diff
@@ -56,17 +56,13 @@ public abstract class ApplicationOptions
...
-        /// <item><description><c>urn:ietf:wg:oauth:2.0:oob</c> for desktop (.NET Framework and .NET Core) applications</description></item>
-        /// <item><description><c>msal{ClientId}</c> for Xamarin iOS and Xamarin Android without broker (as this will be used by the system web browser by default on these
-        /// platforms to call back the application)
-        /// </description></item>
-        /// </list>
-        /// These default URIs could change in the future.
+        /// This redirect URI needs to be registered in the app registration. See https://aka.ms/msal-net-register-app) for
+        /// details on which redirect URI are defined by default by MSAL.NET for how to register them.
+        /// Also use: <see cref="PublicClientApplicationBuilder.WithDefaultRedirectUri"/> which provides
+        /// good default for public client applications for all platforms.
```
Comment (mmacy) [inline src/client/Microsoft.Identity.Client/AppConfig/ApplicationOptions.cs:62]:
```suggestion
        /// This redirect URI needs to be registered in the app registration. See https://aka.ms/msal-net-register-app for
        /// details on which redirect URIs are defined by default by MSAL.NET and how to register them.
        /// Also use: <see cref="PublicClientApplicationBuilder.WithDefaultRedirectUri"/> which provides
        /// a good default for public client applications for all platforms.
```

File: src/client/Microsoft.Identity.Client/AppConfig/ApplicationOptions.cs [hunk 2]
```diff
@@ -56,17 +56,13 @@ public abstract class ApplicationOptions
...
         ///
-        /// For Web Apps and Web APIs, the redirect URI can be the URL of the application
+        /// For Web Apps and Web APIs, the redirect URI is computed from the URL where the application is running
+        /// (for instance <c>baseUrl//signin-oidc</c> for ASP.NET Core web apps.
```
Comment (mmacy) [inline src/client/Microsoft.Identity.Client/AppConfig/ApplicationOptions.cs:65]:
```suggestion
        /// For web apps and web APIs, the redirect URI is computed from the URL where the application is running
        /// (for instance, <c>baseUrl//signin-oidc</c> for ASP.NET Core web apps).
```

**Review by mmacy — COMMENTED** [review]
Couple of suggested changes, @jmprieur.

**Review by jennyf19 — APPROVED** [review]

**Comment by jennyf19** [comment]
@jmprieur going to merge and fix build in a few.

**Pull request merged**